src/panels/lovelace/cards/hui-map-card.ts

```diff
@@ -182,33 +181,47 @@ class HuiMapCard extends LitElement implements LovelaceCard {
   }
 
   protected shouldUpdate(changedProps: PropertyValues) {
+    if (changedProps.has("hass") || changedProps.has("_config")) {
+      const sourceEntities = this._getSourceEntities(
+        this.hass.states,
+        this._config
+      );
+      if (
+        !this._sourceEntities ||
+        sourceEntities.toString() !== this._sourceEntities.toString()
+      ) {
+        this._sourceEntities = sourceEntities;
+        return true;
+      }
+    }
+
+    // Update if anything other than "hass" has changed.
     if (!changedProps.has("hass") || changedProps.size > 1) {
       return true;
     }
 
     const oldHass = changedProps.get("hass") as HomeAssistant | undefined;
-
-    if (!oldHass || !this._configEntities) {
+    if (!oldHass) {
       return true;
     }
 
-    if (oldHass.themes.darkMode !== this.hass.themes.darkMode) {
+    if (
+      oldHass.themes.darkMode !== this.hass.themes.darkMode ||
+      hasConfigChanged(this, changedProps)
+    ) {
       return true;
     }
 
-    if (changedProps.has("_stateHistory")) {
-      return true;
-    }
+    const entity_ids = (
+      this._configEntities?.map((entity) => entity.entity) || []
+    ).concat(this._sourceEntities || []);
 
-    if (this._config?.geo_location_sources) {
-      if (oldHass.states !== this.hass.states) {
-        return true;
-      }
-    }
-
-    return this._config?.entities
-      ? hasConfigOrEntitiesChanged(this, changedProps)
-      : hasConfigChanged(this, changedProps);
+    return entity_ids.some(
+      (entity_id) =>
+        oldHass.states[entity_id] !== this.hass.states[entity_id] ||
+        oldHass.entities[entity_id]?.display_precision !==
+          this.hass.entities[entity_id]?.display_precision
+    );
```

Comment:
Why this change?

Reply:
Revert these lines. 
(Kept the states comparison separate. One for `this._config.entities` and another one for the source entities.)